### ISSUE TITLE
Is there any way to mark message as inappropriate?

### ISSUE THREAD

**Issue by ARyaskov** [jointwt/twtxt#344]
Hi! Thank you for awesome project!

I've two questions about its web interface:
1) Is there any way to mark any message as inappropriate (with subsequent community manager notification sending)?
2) Is there the community manager role in system (for deleting messages that don't meet community rules)?

Thank you!

**Comment by prologic** [comment]
Hello ARyaskov,

Thank you for your questions. I will answer them both at the same time.

The answer is basically "no" to both. **However:**

Please have a read of the default abuse policy which you can read on my pod (twtxt.net) at https://twtxt.net/abuse

I'm not in favor of moderation / censorship tools. As such we have two mechanisms to deal with abuse:

- Users are able to "Mute" feeds they won't want to see.
- Users are able to report a feed as "Abuse".

In the 2nd case a report goes to the Pod Owner (_configured via `ADMIN_USER` and `ADMIN_EMAIL` via the `twtd` daemon_). It is then up to the Pod Owner/Operator to deal with the abuse report. They then have two options:

- Talk to the person in question about what they've done to see if their actions can be corrected.
  - If so, the offending user has the option to "Edit" or "Delete" their last Twt.
- Delete the offending accounts/feeds.

We basically take a rather harsh and very black/white line here. Basically the Abuse policy was written and adapted from many common sources and basically in a nutshell says, don't abuse others, don't incite violence, don't do illegal activities, don't harrass others and don't expose personal information on others.

Does this make sense? Does this answer your question?

And finally do you have any concerns?

**Comment by prologic** [comment]
One thing I _should_ mention is this abuse policy is just the default baked into the software and these are the tools on offer at the moment.

The stance/posture I take is mostly only related to the following places:

- [twtxt.net](https://twtxt.net) -- My persona pod, the first one that got spun up. It also has quite a few friends, family and colleagues and has become a small community of its own.
- [Twt.social](https://twt.social) -- Fully managed, hosted pods that are managed by me (_at the time of writing this_) which **are** bound by the default abuse policy as stated above.

----

If you run your own pod using this software, then you are welcome to set about any rules/policies you want. Although I _would_ advice (_and this is **NOT** legal adcise_) that you follow the rules and regulations of your local country, given the climate on social media these days (_not great_).

Hope this helps!

TL;DR: I don't tolerate assholary :)

**Comment by ARyaskov** [comment]
Thank you for detailed answer, James! 

It's good to have Mute/Abuse functionality, I didn't notice it when I looked for first time. That's what I searching for.

Just a little proposal - I think it would be better if we had Mute button near each username in Discover section since when twtxt will be big, it will be easier to filter not interesting channels while scrolling down.

**Comment by prologic** [comment]
> Thank you for detailed answer, James! 
> 
> 
> 
> It's good to have Mute/Abuse functionality, I didn't notice it when I looked for first time. That's what I searching for.

You're welcome! Hopefully these two tools and the posture this project is taking is enough because I really really don't want to get into the mess of moderation or censorship, I think it's a terrible idea and some thing I don't think any one person or group of persons have the right to do in my opinion.

Breaking local laws however is an entirely different thing and much easier to reason about. Again in my opinion.
￼￼
> Just a little proposal - I think it would be better if we had Mute button near each username in Discover section since when twtxt will be big, it will be easier to filter not interesting channels while scrolling down.

Actually that's a good idea! It would help to be able to filter through the local pods timeline to find things that might be interesting to you when looking for new feeds to follo. What about just a temporary hide button?￼

**Comment by ARyaskov** [comment]
> Actually that's a good idea! It would help to be able to filter through the local pods timeline to find things that might be interesting to you when looking for new feeds to follo. What about just a temporary hide button?￼

Temporary hide button is pretty cool thing! 
Also we can mute/hide twts _visually_ by dimming (css opacity: 15%) div-element of specified user twts!

**Comment by prologic** [comment]
I like that as it's temporary can be done entirely client side! 👌
